common/arg.cpp

```diff
@@ -57,12 +57,18 @@ static std::string read_file(const std::string & fname) {
 }
 
 static void write_file(const std::string & fname, const std::string & content) {
```

Comment:
if anything here fails, should we try to delete the `(file).tmp` ?

Reply:
I guess... It was tempting to consolidate on fopen within this file too, but we can use both streams and fopen no big deal.